beacon_node/beacon_chain/src/historical_blocks.rs

```diff
@@ -179,51 +325,11 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                 break;
             }
         }
-        // these were pushed in reverse order so we reverse again
+        // These were pushed in reverse order so we reverse again.
         signed_blocks.reverse();
 
-        // Verify signatures in one batch, holding the pubkey cache lock for the shortest duration
-        // possible. For each block fetch the parent root from its successor. Slicing from index 1
-        // is safe because we've already checked that `blocks_to_import` is non-empty.
-        let sig_timer = metrics::start_timer(&metrics::BACKFILL_SIGNATURE_TOTAL_TIMES);
-        let setup_timer = metrics::start_timer(&metrics::BACKFILL_SIGNATURE_SETUP_TIMES);
-        let pubkey_cache = self
-            .validator_pubkey_cache
-            .try_read_for(PUBKEY_CACHE_LOCK_TIMEOUT)
-            .ok_or(HistoricalBlockError::ValidatorPubkeyCacheTimeout)?;
-        let block_roots = signed_blocks
-            .get(1..)
-            .ok_or(HistoricalBlockError::IndexOutOfBounds)?
-            .iter()
-            .map(|block| block.parent_root())
-            .chain(iter::once(anchor_info.oldest_block_parent));
-        let signature_set = signed_blocks
-            .iter()
-            .zip_eq(block_roots)
-            .filter(|&(_block, block_root)| (block_root != self.genesis_block_root))
-            .map(|(block, block_root)| {
-                block_proposal_signature_set_from_parts(
-                    block,
-                    Some(block_root),
-                    block.message().proposer_index(),
-                    &self.spec.fork_at_epoch(block.message().epoch()),
-                    self.genesis_validators_root,
-                    |validator_index| pubkey_cache.get(validator_index).cloned().map(Cow::Owned),
-                    &self.spec,
-                )
-            })
-            .collect::<Result<Vec<_>, _>>()
-            .map_err(HistoricalBlockError::SignatureSet)
-            .map(ParallelSignatureSets::from)?;
-        drop(pubkey_cache);
-        drop(setup_timer);
-
-        let verify_timer = metrics::start_timer(&metrics::BACKFILL_SIGNATURE_VERIFY_TIMES);
-        if !signature_set.verify() {
-            return Err(HistoricalBlockError::InvalidSignature);
-        }
-        drop(verify_timer);
-        drop(sig_timer);
+        // Verify signatures in a batch.
+        self.verify_available_block_signatures(&signed_blocks)?;
```

Comment:
Order of ops should be:
- verify block signatures
- match block and sidecar headers
- verify kzg proofs